src/classes/GRER_SyncGradeLevel_TDTM.cls

```diff
@@ -58,22 +61,53 @@ global with sharing class GRER_SyncGradeLevel_TDTM extends hed.TDTM_Runnable {
         //Declare and initiate lists to update
         List<Contact> contactsToUpdate = new List<Contact>();
         List<Grade_Enrollment__c> gradeEnrolsToUpdate = new List<Grade_Enrollment__c>();
+        
+        //Declare and initiate oldMap
+        Map<ID, Grade_Enrollment__c> oldMap = new Map<ID, Grade_Enrollment__c>();
+        if (listOld != NULL) {
+            oldMap = new Map<ID, Grade_Enrollment__c>((List<Grade_Enrollment__c>)listOld);
+        }
+        //Declare new map to capture Contacts and their related Grade Enrollments
+        Map<ID, List<Grade_Enrollment__c>> contactToGradeEnroll = new Map<ID, List<Grade_Enrollment__c>>();
+             
+        //Declare and initiate set of Grade Enrollment Ids that were changed from a stauts of 'Current'
+        Set<Id> formerGradeEnrolIds = new Set<Id>();
+        
+        //Declare and initiate set of related Contact Ids that had its Grade Enrollments changed from a stauts of 'Current'
+        Set<Id> contactsIdsFormer = new Set<Id>(); 
+            
+        //Declare and initiate lists to show errors on Grade Enrollment records
+        List<Grade_Enrollment__c> gradeEnrolsErrorRecords = new List<Grade_Enrollment__c>();   
+         
+        // Declare and initiate map of Contact Id to its Current Grade Level
+        Map<Id, String> contactIdToCurrentGradeLevel = new Map<Id, String>(); 
 
         //After INSERT and After UPDATE
         if(triggerAction == hed.TDTM_Runnable.Action.AfterInsert 
```

Comment:
I am wondering if we can split AfterInsert and AfterUpdate. We are adding the record ids to list and processing them at last anyway.

Reply:
Hmm I think that was part of your code which needed to run in both AfterInsert and AfterUpdate. So I added my part as an else if {}.

Reply:
We can still split the AfterInsert and AfterUpdate and add a new method for reuse.